ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -584,6 +586,10 @@ let update_allowed_operations ~__context ~self =
   if Db.is_valid_ref __context appliance then
     Xapi_vm_appliance_lifecycle.update_allowed_operations ~__context ~self:appliance
 
+let checkpoint_in_progress ~__context ~vm =
+  Listext.List.setify (List.map snd (Db.VM.get_current_operations ~__context ~self:vm))
```

Comment:
We can keep it but I don't see why `setify` is required. If the same item is multiple times in the list, `List.mem` will return the same result and overall it would be more efficient.